.github/workflows/Tests.yaml

```diff
@@ -440,88 +440,89 @@ jobs:
+          - "gcc-9"
+          - "clang-6.0"
+          - "clang-10"
         include:
-          - sde_arch: -nhm
-            arch_override: nehalem
-          - sde_arch: -snb
-            arch_override: sandybridge
-          - sde_arch: -hsw
-            arch_override: haswell
-          - sde_arch: -skl
-            arch_override: skylake
-          - sde_arch: -icx
-            arch_override: icelake-server
-          - sde_arch: -tgl
-            arch_override: tigerlake
+          - sde_arch: ("-nhm;nehalem" "-snb;sandybridge" "-hsw;haswell"
+              "-skl;skylake")
+            compiler: gcc-7
+          - sde_arch: ("-nhm;nehalem" "-snb;sandybridge" "-hsw;haswell"
+              "-skl;skylake" "-icx;icelake-server")
+            compiler: gcc-8
+          - sde_arch: ("-nhm;nehalem" "-snb;sandybridge" "-hsw;haswell"
+              "-skl;skylake" "-icx;icelake-server")
+            compiler: gcc-9
+          - sde_arch: ("-nhm;nehalem" "-snb;sandybridge" "-hsw;haswell"
+              "-skl;skylake")
+            compiler: clang-6.0
+          - sde_arch: ("-nhm;nehalem" "-snb;sandybridge" "-hsw;haswell"
+              "-skl;skylake" "-icx;icelake-server")
+            compiler: clang-8
+          - sde_arch: ("-nhm;nehalem" "-snb;sandybridge" "-hsw;haswell"
+              "-skl;skylake" "-icx;icelake-server")
+            compiler: clang-9
+          - sde_arch: ("-nhm;nehalem" "-snb;sandybridge" "-hsw;haswell"
+              "-skl;skylake" "-icx;icelake-server" "-tgl;tigerlake")
             compiler: clang-10
     container:
       image: moxcodes/spectrebuildenv:archtests
       env:
         # See the unit test job for the reasons for these configuration choices
         CXXFLAGS: "-Werror"
-        CCACHE_DIR: /work/ccache
-        CCACHE_MAXSIZE: 400M
-        CCACHE_COMPRESS: 1
-        CCACHE_COMPRESSLEVEL: 6
-        CCACHE_COMPILERCHECK: content
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
-      - name: Configure build with cmake
+      - name: Configure, build, and run tests
         working-directory: /work
         # Notes on the build configuration:
         # - We don't need debug symbols during CI, so we turn them off to reduce
         #   memory usage.
         run: >
-          mkdir build && cd build
+          ARCH_PARAM_LIST=${{ matrix.sde_arch }}
 
-          if [[ ${{ matrix.compiler }} =~ (gcc|clang)-([0-9\.]+) ]]; then
-            CC=${BASH_REMATCH[1]}-${BASH_REMATCH[2]};
-            CHARM_CC=${BASH_REMATCH[1]};
-            if [[ ${BASH_REMATCH[1]} = gcc ]]; then
-              CXX=g++-${BASH_REMATCH[2]};
-              FC=gfortran-${BASH_REMATCH[2]};
-            else
-              CXX=clang++-${BASH_REMATCH[2]};
-              FC=gfortran-8;
+          for ARCH_PARAM in ${ARCH_PARAM_LIST[@]}; do
+            OVERRIDE_ARCH=`echo ${ARCH_PARAM} | cut -d";" -f2`
+            SDE_FLAG=`echo ${ARCH_PARAM} | cut -d";" -f1`
+            echo $OVERRIDE_ARCH
```

Comment:
Yeah, this is a pretty quick way to understand which of the archs failed, so I thought it would be good to keep. I've added the extra information you suggested.